### ISSUE TITLE
Auto-lock after x minutes of inactivity

### ISSUE THREAD

**Issue by hpoul** [authpass/authpass#14]

**Comment by hpoul** [comment]
for desktop, probably requires https://github.com/flutter/flutter/issues/30735

**Comment by Jaideep-C** [comment]
"Auto-lock after x minutes of inactivity" Is this issue applicable for platform-android?

**Comment by hpoul** [comment]
@Jaideep-C yep, it is for all platforms. The basic issue is that right now the password database remains open in the background as long as the OS does not kill the app.. i guess we could split this issue into two, separating desktop and mobile.. but the basic steps would be

1. A new preferences setting for Autolock - which can be Never (ie, Off), Immediately when moved into background, after 1 minute, 5 minutes, 10 minutes.
2. Register app life cycle listener to check for app being moved into background
3. After the specified amount of time "lock" files
4. Make sure when the app comes back into the foreground that all lock timers are cleared in case the user switches into to so before the timeout
5. We would need to test this really good, both on ios and android. And make sure it does not interfere with desktop (probably only register lifecycle listeners on mobile?)

Hope this makes sense?

**Comment by Jaideep-C** [comment]
@hpoul Yah that totally made sense!
I have an idea may be this is not the optimistic one but I think we can have a physical notification as soon as the app goes to background till 'X' mins. With that physical physical notification we can reset the timer if the app comes to foreground before the completion of 'X' mins.
Please correct me if there are any limitations or anything wrong in this idea!

**Comment by hpoul** [comment]
I'm not sure if that is really necessary.. maybe as an optional add-on if someone wants a reminder.. the only useful function i could imagine is a notification like "password files will unlock in x minutes" with a [ lock mow ] button..

> With that physical physical notification we can reset the timer if the app comes to foreground before the completion of 'X' mins.

Couldn't we do that anyway, no matter if there is a notification or not?

Or do you mean iOS scheduled notifications? 🤔

**Comment by gladel** [comment]
Perhaps auto close all files at specified interval?

**Comment by kashvigarg** [comment]
if this isnt closed, id like to learn and contribute to this, help me out here.

**Comment by soumyaDghosh** [comment]
Hi, I would like to work on this issue.
